implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs

```diff
@@ -1,185 +1,351 @@
-                Response::ok(req.id()).body(TransportStatus::new(api_transport, tid))
+                Response::ok(req.id()).body(TransportStatus::new(api_transport))
             }
             Err(msg) => {
                 error!("{}", msg.to_string());
-                Response::bad_request(req.id()).body(TransportStatus::new(
-                    ApiTransport {
-                        tt,
-                        tm,
-                        socket_address: "0.0.0.0:0000".parse().unwrap(),
-                        worker_address: "<none>".into(),
-                        flow_control_id: None,
-                    },
-                    "<none>".to_string(),
-                ))
+                Response::bad_request(req.id()).body(TransportStatus::new(ApiTransport {
+                    tt: Tcp,
+                    tm: Listen,
+                    socket_address: "0.0.0.0:0000".parse().unwrap(),
+                    worker_address: "<none>".into(),
+                    processor_address: "<none>".into(),
+                    flow_control_id: FlowControls::generate_id(), // FIXME
+                }))
             }
         };
 
         Ok(response)
     }
 
-    pub(super) async fn delete_transport(
+    pub(super) async fn delete_tcp_connection(
         &self,
         req: &Request<'_>,
         dec: &mut Decoder<'_>,
     ) -> Result<ResponseBuilder<()>> {
-        let mut node_manager = self.node_manager.write().await;
+        let node_manager = self.node_manager.read().await;
         let body: DeleteTransport = dec.decode()?;
-        info!("Handling request to delete transport: {}", body.tid);
-
-        let tid: Alias = body.tid.to_string();
-
-        match node_manager.transports.get(&tid) {
-            Some(t) if t.tm == TransportMode::Listen => {
-                // FIXME: stopping the listener shuts down the entire node
-                // node_manager
-                //     .tcp_transport
-                //     .stop_listener(&t.worker_address)
-                //     .await?;
-                node_manager.transports.remove(&tid);
-                Ok(Response::ok(req.id()))
+        info!("Handling request to stop listener: {}", body.address);
+
+        let sender_address = match body.address.parse::<SocketAddr>() {
+            Ok(socket_address) => {
+                match node_manager
+                    .tcp_transport
+                    .registry()
+                    .get_all_sender_workers()
+                    .iter()
+                    .find(|x| x.socket_address() == socket_address)
+                    .map(|x| x.address().clone())
+                {
+                    None => return Ok(Response::bad_request(req.id())),
+                    Some(addr) => addr,
+                }
             }
-            Some(t) => {
-                node_manager
+            Err(_err) => body.address.into(),
+        };
+
+        match node_manager.tcp_transport.disconnect(&sender_address).await {
```

Comment:
Is it OK not to remove the tcp transport from the registry?

Reply:
It will be removed automatically when Workers are shutdown